### PR TITLE
CodeGen/X86: add missing mcpu to tests

### PR DESCRIPTION
**Pull request by artagnon**
In order to faciliate working on core CodeGen infrastructure that improves the schedule of instructions in the default case where there is no scheduling information available in the machine model, add the missing -mcpu to existing tests, so that future patches doesn't unnecessarily change a huge number of tests. Note that the -mcpu argument is already present on all tests under most targets, including ARM and AArch64 targets.

Care has been taken to pick the right mcpu for each test, after studying its intent, and the patch should only affect the schedule of instructions in each test.

### CODE REVIEW

File: llvm/test/CodeGen/X86/bitcast-and-setcc-256.ll [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
-; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mattr=+SSE2 | FileCheck %s --check-prefix=SSE2-SSSE3
-; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mattr=+SSSE3 | FileCheck %s --check-prefix=SSE2-SSSE3
-; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mattr=+avx | FileCheck %s --check-prefixes=AVX12,AVX1
-; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mattr=+avx2 | FileCheck %s --check-prefixes=AVX12,AVX2
-; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mattr=+avx512f,+avx512vl | FileCheck %s --check-prefix=AVX512F
-; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mattr=+avx512f,+avx512vl,+avx512bw | FileCheck %s --check-prefix=AVX512BW
+; RUN: llc < %s -mtriple=x86_64-unknown-unknown -mcpu=x86-64 -mattr=+SSE2 | FileCheck %s --check-prefix=SSE2-SSSE3
```
Comment (arsenm) [inline llvm/test/CodeGen/X86/bitcast-and-setcc-256.ll:2]: I don't know the details of x86, but I'd assume this is the default cpu anyway

File: llvm/test/CodeGen/X86/fold-tied-op.ll [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
-; RUN: llc -verify-machineinstrs -mtriple=i386--netbsd < %s | FileCheck %s
+; RUN: llc -verify-machineinstrs -mtriple=i386--netbsd -mcpu=generic < %s | FileCheck %s
```
Comment (arsenm) [inline llvm/test/CodeGen/X86/fold-tied-op.ll:2]: I'd assume this is the default? What are the default cpus?

**Comment by RKSimon** [comment]
The default x86 cpu / tuning attributes are rather messy - I'm not sure hacking the tests is the best way to solve this

**Comment by artagnon** [comment]
> The default x86 cpu / tuning attributes are rather messy - I'm not sure hacking the tests is the best way to solve this

Yes, I realized that when updating the tests. However, would we rather have a patch that's improving the schedule in the fallback case generate a ton of test changes?

File: llvm/test/CodeGen/X86/bitcast-and-setcc-256.ll (hunk 1 above)
Comment (artagnon) [reply llvm/test/CodeGen/X86/bitcast-and-setcc-256.ll:2]: I'm trying to pick an mcpu that has scheduling information for the instructions that are used by the test. I checked that #101389 doesn't generate any additional changes after this patch.

**Comment by phoebewang** [comment]
> The default x86 cpu / tuning attributes are rather messy

IIRC, we intend to use early target as default cpu and popular target as default tuning, so that we can get the both compatibility and performance.

> However, would we rather have a patch that's improving the schedule in the fallback case generate a ton of test changes?

Can we disable scheduling for these test as a beginning?

**Comment by artagnon** [comment]
> Can we disable scheduling for these test as a beginning?

`-enable-misched=false` doesn't solve the problem of patches generating test updates.

**Comment by artagnon** [comment]
Not pursuing this.

**Pull request closed without merging**